sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/RestfulClient.java

```diff
@@ -0,0 +1,187 @@
+      String keyStorePath) {
+    final HttpURLConnection connection;
+
+    try {
+      if (Strings.isNullOrEmpty(keyStorePath)) {
+        connection = (HttpURLConnection) new URL(urlStr).openConnection();
+      } else {
+        connection = (HttpsURLConnection) new URL(urlStr).openConnection();
+        ((HttpsURLConnection) connection).setSSLSocketFactory(buildSslSocketFactory(keyStorePath));
+      }
+
+      connection.setRequestMethod(httpMethod);
+      connection.setConnectTimeout(TIMEOUT_MILLIS);
+      connection.setReadTimeout(TIMEOUT_MILLIS);
+
+      if (requestPropertyMap != null) {
+        for (Map.Entry<String, String> requestProperty : requestPropertyMap.entrySet()) {
+          connection.setRequestProperty(requestProperty.getKey(), requestProperty.getValue());
+        }
+      }
+
+      int responseCode = connection.getResponseCode();
+      if (responseCode != 200) {
+        logger.log(
+            Level.WARNING,
+            String.format(
+                "Error reponse from %s code (%s) text %s",
+                urlStr, responseCode, readResponseString(connection)));
+        return "";
+      }
+
+      return readResponseString(connection).trim();
+
+    } catch (IOException e) {
+      logger.log(Level.WARNING, String.format("Restful Client fetch string failed: %s", e));
+    }
+
+    return "";
+  }
+
+  private static String readResponseString(HttpURLConnection connection) {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    try (InputStream is = connection.getInputStream()) {
+      ByteStreams.copy(is, os);
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try (InputStream is = connection.getErrorStream()) {
+      if (is != null) {
+        ByteStreams.copy(is, os);
+      }
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try {
+      return os.toString(StandardCharsets.UTF_8.name());
+    } catch (UnsupportedEncodingException e) {
+      throw new IllegalStateException("UTF-8 not supported can't happen.", e);
+    }
+  }
+
+  private static SSLSocketFactory buildSslSocketFactory(String keystorePath) {
+    try {
+      KeyStore keyStore = getK8sKeystore(keystorePath);
+
+      String tmfAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(tmfAlgorithm);
+      tmf.init(keyStore);
+
+      SSLContext context = SSLContext.getInstance("TLS");
+      context.init(null, tmf.getTrustManagers(), null);
+      return context.getSocketFactory();
+
+    } catch (Exception e) {
+      logger.log(Level.WARNING, "Build SslSocketFactory for K8s restful client exception: %s", e);
+    }
+    return null;
+  }
+
+  String getK8sCredHeader(String tokenFilePath) {
+    try {
+      File file = new File(tokenFilePath);
+      String content = Files.asCharSource(file, Charsets.UTF_8).read();
+      return String.format("Bearer %s", content);
```

Comment:
```suggestion
      return "Bearer " + content;
```


Avoid `String.format` since it's slow performance and rarely that much more readable

Reply:
ok, will change all of String.format to '+'